### PR TITLE
New onion_message support 2/3: importing the spec and creating helpers

### PR DESCRIPTION
**Pull request by rustyrussell**
(based on #4798)

Changelog-None

### CODE REVIEW

File: common/blindedpath.c [hunk 1]
```diff
@@ -147,38 +147,15 @@ static bool blind_node(const struct privkey *blinding,
 	return true;
 }
 
-bool blinded_node_alias(const struct privkey *blinding,
```
Comment (cdecker) [inline common/blindedpath.c]: Looks like this commit could be merged with the prior one.

File: hsmd/hsmd_wire.csv [hunk 2]
```diff
@@ -21,6 +21,7 @@ msgtype,hsmd_init_reply,111
 msgdata,hsmd_init_reply,node_id,node_id,
 msgdata,hsmd_init_reply,bip32,ext_key,
 msgdata,hsmd_init_reply,bolt12,pubkey32,
+msgdata,hsmd_init_reply,onion_reply_secret,secret,
```
Comment (cdecker) [inline hsmd/hsmd_wire.csv:24]: This is a constant that we could just as well store in the DB.

**Comment by cdecker** [comment]
ACK https://github.com/ElementsProject/lightning/pull/4799/commits/65694bfa77dd6282229c1d2f1882590f4d331168

File: common/blindedpath.c (hunk 1 above)
Comment (rustyrussell) [reply common/blindedpath.c]: Yes, I forgot to fold it in.  Fixing.

File: hsmd/hsmd_wire.csv (hunk 2 above)
Comment (rustyrussell) [reply hsmd/hsmd_wire.csv:24]: We *could*, but it is a secret which can be (ab)used to reveal our identity.  We need to derive it deterministically anyway, so it makes sense to do it here.

**Comment by rustyrussell** [comment]
Trivial rebase (and merged fixup commit) now #4798 merged.

**Pull request merged**